packages/variable-resolver/src/browser/variable-quick-open-service.ts

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (C) 2018 Red Hat, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+
+import { inject, injectable } from 'inversify';
+import { QuickOpenService, QuickOpenModel, QuickOpenItem, QuickOpenMode } from '@theia/core/lib/browser/quick-open/';
+import { VariableRegistry } from './variable';
+
+@injectable()
+export class VariableQuickOpenService implements QuickOpenModel {
+
+    protected items: QuickOpenItem[];
+
+    constructor(
+        @inject(VariableRegistry) protected readonly variableRegistry: VariableRegistry,
+        @inject(QuickOpenService) protected readonly quickOpenService: QuickOpenService
+    ) { }
+
+    open(): void {
+        const quickOpenItems = this.variableRegistry.getVariables().map(
+            v => new VariableQuickOpenItem(v.name, v.description)
+        );
+        this.items = [...quickOpenItems];
```

Comment:
`this.items = quickOpenItems` ? `map` fn creates a new array

Reply:
Sure ) Changed